lib/ts/recipe/thirdpartypasswordless/types.ts

```diff
@@ -0,0 +1,213 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+    UserType,
+    InputType as AuthRecipeInputType,
+    NormalisedInputType as AuthRecipeNormalisedInputType,
+} from "../authRecipeWithEmailVerification/types";
+import {
+    PasswordlessFlowType,
+    PasswordlessUser,
+    PreAndPostAPIHookAction as PasswordlessPreAndPostAPIHookAction,
+} from "../passwordless/types";
+import { RecipeFunctionOptions, RecipePostAPIHookContext, RecipePreAPIHookContext } from "../recipeModule/types";
+import { StateObject, PreAndPostAPIHookAction as ThirdPartyPreAndPostAPIHookAction } from "../thirdparty/types";
+import { InputTypeOverride as EmailVerificationOverride } from "../emailverification/types";
+import OverrideableBuilder from "supertokens-js-override";
+
+export type PreAndPostAPIHookAction = ThirdPartyPreAndPostAPIHookAction | PasswordlessPreAndPostAPIHookAction;
+
+export type PreAPIHookContext = RecipePreAPIHookContext<PreAndPostAPIHookAction>;
+export type PostAPIHookContext = RecipePostAPIHookContext<PreAndPostAPIHookAction>;
+
+export type InputType = AuthRecipeInputType<PreAndPostAPIHookAction> & {
+    override?: {
+        emailVerification?: EmailVerificationOverride;
+        functions?: (
+            originalImplementation: RecipeInterface,
+            builder: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+    };
+};
+
+export type NormalisedInputType = AuthRecipeNormalisedInputType<PreAndPostAPIHookAction> & {
+    override: {
+        functions: (
+            originalImplementation: RecipeInterface,
+            builder: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+    };
+};
+
+export type RecipeInterface = {
+    getAuthorisationURLFromBackend: (input: {
+        providerId: string;
+        userContext: any;
+        options?: RecipeFunctionOptions;
+    }) => Promise<{
+        status: "OK";
+        url: string;
+        fetchResponse: Response;
+    }>;
+
+    thirdPartySignInAndUp: (input: { userContext: any; options?: RecipeFunctionOptions }) => Promise<
+        | {
+              status: "OK";
+              user: UserType;
+              createdNewUser: boolean;
+              fetchResponse: Response;
+          }
+        | {
+              status: "NO_EMAIL_GIVEN_BY_PROVIDER";
+              fetchResponse: Response;
+          }
+    >;
+
+    getStateAndOtherInfoFromStorage: <CustomStateProperties>(input: {
+        userContext: any;
+    }) => (StateObject & CustomStateProperties) | undefined;
+
+    setStateAndOtherInfoToStorage: <CustomStateProperties>(input: {
+        state: StateObject & CustomStateProperties;
+        userContext: any;
+    }) => void;
+
+    getAuthorizationURLWithQueryParamsAndSetState: (input: {
```

Comment:
This is called `Authorization`, but the other function is `Authorisation`. We need to make them consistent everywhere?

Reply:
Changed this everywhere to use `Authorisation`